test/query/test-db.ts

```diff
@@ -6,7 +6,23 @@ type JsonTestTable = PostgresTable<{
   id: number;
   // NOTE: this could be JSON, but this could also be a composite column
   json_column: { someVal: number };
+  nullable_json_column: { nullableVal: number } | null;
+  nested_json_column: {
+    obj: {
+      nestedVal: number;
+      obj: { deeplyNestedVal: number };
+    };
+  };
+  nullable_nested_json_column: {
+    obj: {
+      nullableNestedVal: number;
+      obj: { nullableDeeplyNestedVal: number };
+    };
+  } | null;
   array_composite: number[];
+  nullable_array_composite: number[] | null;
+  array_of_objects: { arrayVal: number }[];
+  nullable_array_of_objects: { nullableArrayVal: number }[] | null;
```

Comment:
There are more combinations with more nesting, but it's difficult to cover everything.